### PR TITLE
Zmz/layer norm1

### PR DESCRIPTION
**Pull request by hellozmz**
## Motivation and Context
impl layer_norm use op_api

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by zhangzefeng92 — APPROVED** [review]

File: impl/ascend_npu/diopi_impl/layer_norm.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+diopiError_t diopiLayerNormBackward(diopiContextHandle_t ctx, diopiTensorHandle_t gradInput, diopiTensorHandle_t gradWeight, diopiTensorHandle_t gradBias,
+                                    diopiConstTensorHandle_t gradOutput, diopiConstTensorHandle_t input, diopiConstTensorHandle_t weight,
+                                    diopiConstTensorHandle_t bias, diopiConstTensorHandle_t mean, diopiConstTensorHandle_t rstd, diopiSize_t normalizedShape) {
+    BEGIN_CALL_ACL_OP(gradInput, gradWeight, gradBias, gradOutput, input, weight, bias, mean, rstd);
+    std::vector<int64_t> normalizedVec(normalizedShape.data, normalizedShape.data + normalizedShape.len);
+    std::array<bool, 3> maskVec{true, true, true};
+    auto result = op_api::native_layer_norm_backward(gradOutputAt, inputAt, normalizedVec, meanAt, rstdAt, weightAt, biasAt, maskVec);
+    if (gradInputAt.defined()) {
```
Comment (POI-WX) [inline impl/ascend_npu/diopi_impl/layer_norm.cpp:31]: 未定义时可以不用做任何处理，本身就是空指针

Comment (hellozmz) [reply impl/ascend_npu/diopi_impl/layer_norm.cpp:31]: 传入为空指针的时候，需要调用impl::aten::buildDiopiTensor()给指针赋值的

Comment (hellozmz) [reply impl/ascend_npu/diopi_impl/layer_norm.cpp:31]: > 未定义时可以不用做任何处理，本身就是空指针

传入为空指针的时候，需要调用impl::aten::buildDiopiTensor()给指针赋值

Comment (hellozmz) [reply impl/ascend_npu/diopi_impl/layer_norm.cpp:31]: > 未定义时可以不用做任何处理，本身就是空指针

你是对的，传入为空指针的时候，是不需要构造并填充数据的。

**Review by POI-WX — APPROVED** [review]

File: impl/ascend_npu/diopi_impl/layer_norm.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#include "op_plugin/OpApiInterface.h"
+
+namespace OP_IMPL_NS {
+
+diopiError_t diopiLayerNorm(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiTensorHandle_t saveMean, diopiTensorHandle_t saveInvstd,
+                            diopiConstTensorHandle_t input, diopiConstTensorHandle_t weight, diopiConstTensorHandle_t bias, diopiSize_t normalizedShape,
+                            double eps) {
+    BEGIN_CALL_ACL_OP(out, saveMean, saveInvstd, input, weight, bias);
+    std::vector<int64_t> normalizedVec(normalizedShape.data, normalizedShape.data + normalizedShape.len);
+    auto result = op_api::native_layer_norm(inputAt, normalizedVec, weightAt, biasAt, eps);
```
Comment (yangbofun) [inline impl/ascend_npu/diopi_impl/layer_norm.cpp:17]: 这里可以手写个out版本，这样少了3次copy。

**Review by yangbofun — APPROVED** [review]

**Comment by hellozmz** [comment]
由于main分支变化，当前代码无法直接跑通测试用例。
目前排查是由于at::Tensor& wrapper__copy_(at::Tensor& self, const at::Tensor& src, bool non_blocking)实现逻辑变化导致。
当前算子不是在internLM中使用，暂缓解决该问题。

**Pull request closed without merging**